app/Community/Components/UserProfileMeta.php

```diff
@@ -0,0 +1,218 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Community\Components;
+
+use App\Community\Enums\Rank;
+use App\Community\Enums\RankType;
+use App\Site\Enums\Permissions;
+use App\Site\Models\User;
+use Illuminate\Contracts\View\View;
+use Illuminate\Support\Carbon;
+use Illuminate\Support\Facades\DB;
+use Illuminate\View\Component;
+
+class UserProfileMeta extends Component
+{
+    private string $averageCompletionPercentage = '0.00';
+    private int $totalHardcoreAchievements = 0;
+    private int $totalSoftcoreAchievements = 0;
+    private User $user;
+    private array $userJoinedGamesAndAwards = [];
+    private array $userMassData = [];
+    private ?array $userClaims = null;
+
+    public function __construct(
+        string $averageCompletionPercentage,
+        User $user,
+        array $userJoinedGamesAndAwards,
+        array $userMassData,
+        ?array $userClaims,
+        int $totalHardcoreAchievements,
+        int $totalSoftcoreAchievements,
+    ) {
+        $this->averageCompletionPercentage = $averageCompletionPercentage;
+        $this->user = $user;
+        $this->userJoinedGamesAndAwards = $userJoinedGamesAndAwards;
+        $this->userMassData = $userMassData;
+        $this->userClaims = $userClaims;
+        $this->totalHardcoreAchievements = $totalHardcoreAchievements;
+        $this->totalSoftcoreAchievements = $totalSoftcoreAchievements;
+    }
```

Comment:
```suggestion
    public function __construct(
        private string $averageCompletionPercentage = '0.00',
        private User $user,
        private array $userJoinedGamesAndAwards = [],
        private array $userMassData = [],
        private ?array $userClaims = null,
        private int $totalHardcoreAchievements = 0,
        private int $totalSoftcoreAchievements = 0,
    ) {
    }
```

Reply:
Thanks, I didn't realize this would work. I'll get into a better habit of doing this -- wonder if there's even a possible linting rule w/ an autofixer? Adjusted in latest.

Reply:
Rector can do such refactoring but it's unfortunately very resource intensive.